File: 0-prep/prepApp/finish.js
const replace = require('replace-in-file');
const {shell_cmd} = require('./shell_cmd')

function printSuccessMessage(projectId, oauthClientID) {
    console.log('\x1b[32m%s\x1b[0m','***Step 1 finished SUCCESSFULLY!!!')
    console.info('\x1b[32m%s\x1b[0m','** DOMAIN WIDE DELEGATION INFO **')
    console.info('\x1b[34m%s\x1b[0m','Go to this URL: https://admin.google.com/u/6/ac/owl/domainwidedelegation')
    console.info('\x1b[33m%s\x1b[0m','Click Add new and use the values below:')
    console.info('\x1b[32m%s\x1b[0m','client ID: ' + oauthClientID)
    console.info('\x1b[32m%s\x1b[0m','OAuth scopes: https://www.googleapis.com/auth/admin.directory.group')
    pauseForSecondStep(projectId)
}

function pauseForSecondStep(projectId) {
    const prompt = require('prompt-sync')();

    const response = prompt('Please enable Domain wide delegation, then type \'y\' to continue\n');
    if(response == 'y' || response == 'Y') {

        const options = {
            files: '../createGroups/.env',
            from: /ADMIN_PROJECT_ID=CHANGE_ME/,
            to: 'ADMIN_PROJECT_ID='+projectId
        }

        replace(options).then(result => {
            const callScript = new shell_cmd()
            callScript.execCommand('pip install -r ../createGroups/requirements.txt && python ../createGroups/create_groups.py').then(res => {
                console.log('\x1b[32m%s\x1b[0m', 'Group creation script ran successfully. Finish the prep by filling out the needed values in helper_scripts/0.5-prep.sh and running it')
                return
                // updateRegion() 
            }).catch(err => {
                console.log('Group creation script had an error', err)
            })
        }).catch(err => {
            console.log('error with replace',err)
        })

    }else{
        console.log('Response did not equal "y", quitting now. You can re-run this script when you are ready. It is idempotent')
    }
}

module.exports = {printSuccessMessage}

File: 0-prep/finish.py
import subprocess
import os
import typer
import shlex
from helpers import *


def success_message(project_id, clientId):
    # print_colored("Step 1 finished successfully", "grn")
    # print_colored("DOMAIN WIDE DELEGATION INFO", "prpl")
    print("DOMAIN WIDE DELEGATION INFO:")
    print("Go to the URL below")

    # print_colored(
    #     "Go to this url https://admin.google.com/u/6/ac/owl/domainwidedelegation", "prpl")
    print("https://admin.google.com/u/6/ac/owl/domainwidedelegation")
    print("Click Add New and use the values below:")
    # print_colored("Client ID: " + clientId, "yllw")
    print("Client ID: " + clientId)

    # print_colored("OAuth scopes: https://www.googleapis.com/auth/admin.directory.group", "yllw")
    print("OAuth scopes: https://www.googleapis.com/auth/admin.directory.group")
    pause_for_second_step(project_id)


def pause_for_second_step(project_id):
    dwd_confirm = typer.confirm(
        "Once you've completed the domain wide delegation steps type Y to continue")
    if(dwd_confirm):
        try:
            cwd = os.getcwd()
            env_file = open(cwd + "/createGroups/.env", "r")
            lines = env_file.readlines()
            for line in lines:
                print(line)
                if("ADMIN_PROJECT_ID" in line):
                    create_groups()
                    return
            print("addming admin project id to env file")
            env_file = open(cwd + "/createGroups/.env", "a")
            env_file.write("\nADMIN_PROJECT_ID="+project_id)
            create_groups()
        except Exception as e:
            print("error creating groups", e)


def create_groups():
    create_groups_cmd = "pip install -r ./createGroups/requirements.txt && python ./createGroups/create_groups.py"
    try:
        subprocess.run(shlex.split(create_groups_cmd), shell=True)
        print("Groups created! Run 0.5-prep to finish prepping your environment")
    except Exception as e:
        print("error on create_groups", e)
